test/test_eos_unit.cpp

```diff
@@ -64,6 +66,38 @@ constexpr Real ev2k = 1.160451812e4;
 #endif // SINGULARITY_TEST_SESAME
 #endif // SPINER_USE_HDF
 
+template <typename E1, typename E2>
+inline void compare_two_eoss(E1 &&test_e, E2 &&ref_e) {
+  // compare all individual member functions with 1 as inputs,
+  // this function is meant to catch mis-implementations of
+  // modifiers that can be initialized in such a way as to
+  // be equivalent of an unmodified eos. Best used with analytic
+  // eoss.
+  REQUIRE(isClose(test_e.TemperatureFromDensityInternalEnergy(1, 1),
+                  ref_e.TemperatureFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.InternalEnergyFromDensityTemperature(1, 1),
+                  ref_e.InternalEnergyFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.PressureFromDensityInternalEnergy(1, 1),
+                  ref_e.PressureFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.SpecificHeatFromDensityInternalEnergy(1, 1),
+                  ref_e.SpecificHeatFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.BulkModulusFromDensityInternalEnergy(1, 1),
+                  ref_e.BulkModulusFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.GruneisenParamFromDensityInternalEnergy(1, 1),
+                  ref_e.GruneisenParamFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.PressureFromDensityTemperature(1, 1),
+                  ref_e.PressureFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.SpecificHeatFromDensityTemperature(1, 1),
+                  ref_e.SpecificHeatFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.BulkModulusFromDensityTemperature(1, 1),
+                  ref_e.BulkModulusFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.GruneisenParamFromDensityTemperature(1, 1),
+                  ref_e.GruneisenParamFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.MinimumDensity(), ref_e.MinimumDensity(), 1.e-15));
+  REQUIRE(isClose(test_e.MinimumTemperature(), ref_e.MinimumTemperature(), 1.e-15));
+  return;
+}
+
 SCENARIO("Test that fast logs are invertible and run on device", "[FastMath]") {
```

Comment:
It looks like a bunch of these tests have nothing to do with the fast logarithms even if they're grouped there. I'm going to create an issue to break these out